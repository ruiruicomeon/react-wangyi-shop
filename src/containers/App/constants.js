const PREFIX = 'APP/';

export const LOAD_APP = PREFIX + 'LOAD_APP';

