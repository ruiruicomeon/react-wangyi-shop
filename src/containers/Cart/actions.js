import {
  ADD_TO_CART,
  DEL_FROM_CART,
  CHECK_CART,
  CHANGE_CART_QUANTITY,
  TRANSFER_TO_USER_CART,
  LOAD_CARTS,
  SET_CARTS,
  SET_IS_CHECK_ALL,
  SET_IS_CHECK_ONE,
  CHECK_ALL,
  UN_CHECK_ALL,
} from './constants';

export function addToCart(product) {
  return {
    type: ADD_TO_CART,
    payload: { product },
  };
}

export function changeCartQuantity(product) {
  return {
    type: CHANGE_CART_QUANTITY,
    payload: { product },
  };
}

export function checkCart(product) {
  return {
    type: CHECK_CART,
    payload: { product },
  };
}

export function delFromCart(product) {
  return {
    type: DEL_FROM_CART,
    payload: { product },
  };
}

export function transferToUserCart() {
  return {
    type: TRANSFER_TO_USER_CART,
  };
}

export function loadCarts() {
  return {
    type: LOAD_CARTS,
  };
}

export function setCarts(carts) {
  return {
    type: SET_CARTS,
    payload: { carts },
  };
}

export function setIsCheckAll(isCheckAll) {
  return {
    type: SET_IS_CHECK_ALL,
    payload: { isCheckAll },
  };
}

export function setIsCheckOne(isCheckOne) {
  return {
    type: SET_IS_CHECK_ONE,
    payload: { isCheckOne },
  };
}

export function checkAll() {
  return {
    type: CHECK_ALL,
  };
}

export function unCheckAll() {
  return {
    type: UN_CHECK_ALL,
  };
}
