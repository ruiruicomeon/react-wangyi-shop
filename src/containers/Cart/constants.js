const PREFIX = 'CART/';

export const ADD_TO_CART = PREFIX + 'ADD_TO_CART';

export const CHANGE_CART_QUANTITY = PREFIX + 'CHANGE_CART_QUANTITY';

export const CHECK_CART = PREFIX + 'CHECK_CART';

export const DEL_FROM_CART = PREFIX + 'DEL_FROM_CART';

export const TRANSFER_TO_USER_CART = PREFIX + 'TRANSFER_TO_USER_CART';

export const LOAD_CARTS = PREFIX + 'LOAD_CARTS';

export const SET_CARTS = PREFIX + 'SET_CARTS';

export const SET_IS_CHECK_ALL = PREFIX + 'SET_IS_CHECK_ALL';

export const SET_IS_CHECK_ONE = PREFIX + 'SET_IS_CHECK_ONE';

export const CHECK_ALL = PREFIX + 'CHECK_ALL';

export const UN_CHECK_ALL = PREFIX + 'UN_CHECK_ALL';
