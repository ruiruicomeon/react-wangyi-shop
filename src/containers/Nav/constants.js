const PREFIX = 'NAV/';

export const FETCH_NOTICES = PREFIX + 'FETCH_NOTICES';

export const SET_NOTICES = PREFIX + 'SET_NOTICES';
