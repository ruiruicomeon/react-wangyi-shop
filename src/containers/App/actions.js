import { LOAD_APP } from './constants';

export function loadApp() {
  return {
    type: LOAD_APP,
  };
}
