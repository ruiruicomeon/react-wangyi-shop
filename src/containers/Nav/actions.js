import { FETCH_NOTICES, SET_NOTICES } from './constants';

export function fetchNotices() {
  return {
    type: FETCH_NOTICES,
  };
}

export function setNotices(notices) {
  return {
    type: SET_NOTICES,
    payload: {
      notices,
    },
  };
}
