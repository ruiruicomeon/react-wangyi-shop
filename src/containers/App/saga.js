import { call, put, takeLatest } from 'redux-saga/effects';
import _ from 'lodash';

import { LOAD_APP } from './constants';
import { loadCarts } from '@/containers/Cart/actions';
import { setUser } from '@/containers/Login/actions';
import { tokenStorage } from '@/utils/localStorage';
import { authApi, api } from '@/api';

export function* loadApp() {
  try {
    let token = tokenStorage.load();

    api.init();
    api.setToken(token);
    const res = yield call(authApi.user);

    const user = _.get(res, 'data.user', null);

    if (!_.isEmpty(user)) {
      token = user.token;
      api.setToken(token);
      tokenStorage.save(token);
    }

    yield put(setUser(user));
    yield put(loadCarts());
  } catch (err) {
    console.error('校验错误: ', err.response || err);
    api.setToken(null);
    tokenStorage.save(null);
    yield put(setUser(null));
    yield put(loadCarts());
  }
}

export default function* saga() {
  yield takeLatest(LOAD_APP, loadApp);
}
