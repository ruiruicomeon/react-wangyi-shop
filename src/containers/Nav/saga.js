import { call, put, takeLatest } from 'redux-saga/effects';
import _ from 'lodash';

import { FETCH_NOTICES } from './constants';
import { setNotices } from './actions';
import { noticeApi } from '@/api';

export function* getNotices() {
  try {
    const res = yield call(noticeApi.getNotices);
    const notices = _.get(res, 'data.notices', []);

    yield put(setNotices(notices));
  } catch (err) {
    console.error('获取通知错误: ', err.response || err);
  }
}

export default function* saga() {
  yield takeLatest(FETCH_NOTICES, getNotices);
}
