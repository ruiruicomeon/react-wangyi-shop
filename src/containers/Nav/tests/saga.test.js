/**
 * Tests for HomePage sagas
 */

import { put, takeLatest } from 'redux-saga/effects';

import { FETCH_NOTICES } from '../constants';
import { setNotices } from '../actions';
import saga, { getNotices } from '../saga';

describe('getNotices Saga', () => {
  let getReposGenerator;

  beforeEach(() => {
    getReposGenerator = getNotices();
    const selectDescriptor = getReposGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('获取通知成功后调用 setNotices action', () => {
    const notices = ['hello', 'notice'];
    const response = {
      data: {
        notices,
      },
    };
    const putDescriptor = getReposGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(setNotices(notices)));
  });
});

describe('Saga', () => {
  const navSaga = saga();

  it('saga接受 FETCH_NOTICES action', () => {
    const takeLatestDescriptor = navSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(FETCH_NOTICES, getNotices));
  });
});
