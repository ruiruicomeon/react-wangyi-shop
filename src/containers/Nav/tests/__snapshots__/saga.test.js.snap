// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getNotices Saga 获取通知成功后调用 setNotices action 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
